### ISSUE TITLE
Resin push should ignore `.git` directories, and files pointed to by .gitignore, .dockerignore

### ISSUE THREAD

**Issue by CameronDiver** [balena-io/balena-cli#889]
Often these files are completely useless and add weight (and therefore time) to the source package upload. Ignoring these (maybe behind a flag in case somebody wants their git history deployed) would make the majority of repos much faster to upload and start to build.

<img src="https://frontapp.com/assets/img/icons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_d8o3)

**Comment by richbayliss** [comment]
+1 this makes a lot of sense to me.

**Comment by pimterry** [comment]
Totally agree, it definitely should. Right now I'm not going to find the time for this any time soon, but I'd happily review a PR if somebody else wants to take a look at it :+1:.

**Comment by zrzka** [comment]
Just to add some numbers with simple [rust application](https://github.com/zrzka/rust-thermometer):

```bash
 2018-07-03 17:01:55 ⌚  Zrzka in ~/Work/rust-thermometer
± |master → origin {1} U:4 ✗| → du -sh .
564M	.

 2018-07-03 17:09:15 ⌚  Zrzka in ~/Work/rust-thermometer
± |master → origin {1} U:3 ✗| → du -sh target
562M	target
```

Removing `target` (build artefacts) or move it aside for a while is not an option, because [RLS](https://github.com/rust-lang-nursery/rls) starts immediately building it again when these files are gone. As a workaround for now, one have to clone workspace, move it aside, \[local\] push, move it back and open workspace again. Or wait ... And this is quite simple app, these numbers are much bigger for other projects.

**Comment by pdcastro** [comment]
Implemented in PR #950. Further work required in issue #1032.
